packages/flutter/lib/src/widgets/tooltip.dart

```diff
@@ -0,0 +1,910 @@
+  tap,
+}
+
+/// Signature for when a tooltip is triggered.
+typedef TooltipTriggeredCallback = VoidCallback;
+
+/// A special [MouseRegion] that when nested, only the first [_ExclusiveMouseRegion]
+/// to be hit in hit-testing order will be added to the BoxHitTestResult (i.e.,
+/// child over parent, last sibling over first sibling).
+///
+/// The [onEnter] method will be called when a mouse pointer enters this
+/// [MouseRegion], and there is no other [_ExclusiveMouseRegion]s obstructing
+/// this [_ExclusiveMouseRegion] from receiving the events. This includes the
+/// case where the mouse cursor stays within the paint bounds of an outer
+/// [_ExclusiveMouseRegion], but moves outside of the bounds of the inner
+/// [_ExclusiveMouseRegion] that was initially blocking the outer widget.
+///
+/// Likewise, [onExit] is called when the a mouse pointer moves out of the paint
+/// bounds of this widget, or moves into another [_ExclusiveMouseRegion] that
+/// overlaps this widget in hit-testing order.
+///
+/// This widget doesn't affect [MouseRegion]s that aren't [_ExclusiveMouseRegion]s,
+/// or other [HitTestTarget]s in the tree.
+class _ExclusiveMouseRegion extends MouseRegion {
+  const _ExclusiveMouseRegion({super.onEnter, super.onExit, super.cursor, super.child});
+
+  @override
+  _RenderExclusiveMouseRegion createRenderObject(BuildContext context) {
+    return _RenderExclusiveMouseRegion(onEnter: onEnter, onExit: onExit, cursor: cursor);
+  }
+}
+
+class _RenderExclusiveMouseRegion extends RenderMouseRegion {
+  _RenderExclusiveMouseRegion({super.onEnter, super.onExit, super.cursor});
+
+  static bool isOutermostMouseRegion = true;
+  static bool foundInnermostMouseRegion = false;
+
+  @override
+  bool hitTest(BoxHitTestResult result, {required Offset position}) {
+    var isHit = false;
+    final bool outermost = isOutermostMouseRegion;
+    isOutermostMouseRegion = false;
+    if (size.contains(position)) {
+      isHit = hitTestChildren(result, position: position) || hitTestSelf(position);
+      if ((isHit || behavior == HitTestBehavior.translucent) && !foundInnermostMouseRegion) {
+        foundInnermostMouseRegion = true;
+        result.add(BoxHitTestEntry(this, position));
+      }
+    }
+
+    if (outermost) {
+      // The outermost region resets the global states.
+      isOutermostMouseRegion = true;
+      foundInnermostMouseRegion = false;
+    }
+    return isHit;
+  }
+}
+
+/// A design-language-agnostic tooltip.
+///
+/// A tooltip provides a text label which helps explain the function of a button
+/// or some other user interface action.
+///
+/// [RawTooltip] can be triggered in three ways:
+///  * By a long press gesture.
+///  * By a tap gesture.
+///  * By hovering over the widget with a mouse pointer.
+///  * Manually, using the [RawTooltipState.ensureTooltipVisible] function.
+///
+/// See also:
+///
+///   * [Tooltip], a higher-level widget in the Material library that wraps
```

Comment:
Are there a lot of additional functionalities? Isn't it mostly just with a material design layout by default?